astropy/io/fits/tests/test_core.py

```diff
@@ -1286,6 +1288,34 @@ def test_simulateonly(self):
             hdul[0].writeto(None)
             hdul[0].header.tofile(None)
 
+    def test_bintablehdu_zero_bytes(self):
+        """Make sure we don't have any zero-byte writes in BinTableHDU"""
+
+        bright = np.rec.array([(1, 'Sirius', -1.45, 'A1V'),
+                            (2, 'Canopus', -0.73, 'F0Ib'),
+                            (3, 'Rigil Kent', -0.1, 'G2V')],
+                           formats='int16,a20,float32,a10', names='order,name,mag,Sp')
+
+        hdu_non_zero = fits.BinTableHDU(bright)
+        # use safeio, a special file handler meant to fail on zero-byte writes
+        fh = safeio._BufferedWriter(open('bright.fits', mode='wb'))
+        hdu_non_zero.writeto(fh)
+        fh.close()
+        assert os.stat('bright.fits').st_size == fits.PrimaryHDU().filebytes() + hdu_non_zero.filebytes()
```

Comment:
The size check seems irrelevant here since a zero byte write would not change the size.

Reply:
Ok, I'll remove the size check in both tests and rely on the ValueError from `safeio.CatchZeroByteWriter` to cause the test to fail.

Reply:
The size check was meant to verify that none of the _non-zero_ length writes were skipped by the change. But as noted in https://github.com/astropy/astropy/pull/11955#issuecomment-901451629 anything not padded to the correct block size should fail validation in other places, so probably does not need an extra test here.